tests/GraphBLAS-sharp.Tests/Matrix/Map.fs

```diff
@@ -0,0 +1,141 @@
+
+    "Arrays must be the same"
+    |> Utils.compare2DArrays isEqual actual2D expected2D
+
+let correctnessGenericTest
+    zero
+    op
+    (addFun: MailboxProcessor<_> -> AllocationFlag -> ClMatrix<'a> -> ClMatrix<'b>)
+    toCOOFun
+    (isEqual: 'a -> 'a -> bool)
+    q
+    (case: OperationCase<MatrixFormat>)
+    (matrix: 'a [,])
+    =
+
+    let mtx =
+        Utils.createMatrixFromArray2D case.Format matrix (isEqual zero)
+
+    if mtx.NNZ > 0 then
+        try
+            let m = mtx.ToDevice case.TestContext.ClContext
+
+            let res = addFun q HostInterop m
+
+            m.Dispose q
+
+            let (cooRes: ClMatrix<'a>) = toCOOFun q HostInterop res
+            let actual = cooRes.ToHost q
+
+            cooRes.Dispose q
+            res.Dispose q
+
+            logger.debug (
+                eventX "Actual is {actual}"
+                >> setField "actual" (sprintf "%A" actual)
+            )
+
+            checkResult isEqual op zero matrix actual
+        with
+        | ex when ex.Message = "InvalidBufferSize" -> ()
+        | ex -> raise ex
+
+let createTestMap case (zero: 'a) op isEqual opQ map =
+    let getCorrectnessTestName = getCorrectnessTestName case
+
+    let context = case.TestContext.ClContext
+    let q = case.TestContext.Queue
+
+    let map = map context opQ wgSize
+
+    let toCOO = Matrix.toCOO context wgSize
+
+    case
+    |> correctnessGenericTest zero op map toCOO isEqual q
+    |> testPropertyWithConfig config (getCorrectnessTestName $"{typeof<'a>}")
+
+let testFixturesMapNot case =
+    [ let q = case.TestContext.Queue
+      q.Error.Add(fun e -> failwithf "%A" e)
+
+      createTestMap case false not (=) ArithmeticOperations.notQ Matrix.map ]
+
+let notTests =
+    operationGPUTests "Backend.Matrix.map not tests" testFixturesMapNot
+
+let testFixturesMapAdd case =
+    [ let context = case.TestContext.ClContext
+      let q = case.TestContext.Queue
+      q.Error.Add(fun e -> failwithf "%A" e)
+
+      if Utils.isFloat64Available context.ClDevice then
+          createTestMap case 0.0 ((+) 10.0) Utils.floatIsEqual (ArithmeticOperations.addLeftConst 0.0 10.0) Matrix.map
+
+      createTestMap case 0.0f ((+) 10.0f) Utils.float32IsEqual (ArithmeticOperations.addLeftConst 0.0f 10.0f) Matrix.map
+
+      createTestMap case 0uy ((+) 10uy) (=) (ArithmeticOperations.addLeftConst 0uy 10uy) Matrix.map ]
+
+let addTests =
+    operationGPUTests "Backend.Matrix.map add tests" testFixturesMapAdd
+
+let testFixturesMapMul case =
+    [ let context = case.TestContext.ClContext
+      let q = case.TestContext.Queue
+      q.Error.Add(fun e -> failwithf "%A" e)
+
+      if Utils.isFloat64Available context.ClDevice then
+          createTestMap case 0.0 ((*) 10.0) Utils.floatIsEqual (ArithmeticOperations.mulLeftConst 0.0 10.0) Matrix.map
+
+      createTestMap case 0.0f ((*) 10.0f) Utils.float32IsEqual (ArithmeticOperations.mulLeftConst 0.0f 10.0f) Matrix.map
```

Comment:
We can pass zero, constant and both functions once to the `createTestMap`. So as not to write them twice